### PR TITLE
Extract configuration sections to separate files by topic.

### PR DESCRIPTION
**Pull request by albertodonato**
Trivial change to split configuration sections from the configuration.md doc to different files, one per section.
We'll want to expand each of them with more detailed info about the specific component

### CODE REVIEW

**Comment by stgraber** [comment]
Yay, doc improvements!

Can you make a separate commit for each section? Storage and Network don't exist in 2.0 so that'd make cherry-picking a bit cleaner.

Now for the naming and splitting bikeshedding, I think we should get:
 - configuration.md (just a bunch of links to other documents)
 - containers.md (container config & devices)
 - networks.md (network config)
 - profiles.md (blurb about profiles, inheritance, ... and link to containers.md for config)
 - server.md (server config)
 - storage.md (storage config and current content of storage-backends.md)

@brauner what do you think? ^

I need to look at our website generator to have it pick all that stuff up and see how we can do category tagging in our markdown to have things pulled into the right sections of the website, but splitting up configuration.md should help quite a bit!

**Comment by albertodonato** [comment]
@stgraber Updated as you suggested

**Comment by brauner** [comment]
@stgraber, fine with me. :)

File: doc/configuration.md [hunk 1]
```diff
@@ -1,415 +1,8 @@
...
-zfs.use\_refquota       | string    | zfs driver                | same as volume.zfs.zfs\_requota       | Use refquota instead of quota for space.
-
-Storage volume configuration keys can be set using the lxc tool with:
-
-    lxc storage volume set [<remote>:]<pool> <volume> <key> <value>
+- [the LXD server itself](server.md)
+- [containers](containers.md) 
+- [networking](networks.md)
+- [profiles](profiles.md)
+- [storage pools and backends](storage.md)
```
Comment (brauner) [inline doc/configuration.md]: This is a nit but can we name this:
- `Server`
- `Containers`
- `Networking`
- `Profiles`
- `Storage`

Comment (albertodonato) [inline doc/configuration.md]: Done

**Comment by dann1** [comment]
Ohh yeah, niiice.

**Pull request merged**